packages/flutter/lib/src/material/ink_well.dart

```diff
@@ -775,6 +775,11 @@ class _InkResponseState extends State<_InkResponseStateWidget>
   @override
   void didUpdateWidget(_InkResponseStateWidget oldWidget) {
     super.didUpdateWidget(oldWidget);
+    if (widget.key != oldWidget.key) {
```

Comment:
this is guaranteed to always be false

if you can ever find a case where this is true, we have a serious bug

Reply:
You are right, I will revertPR and rethink about https://github.com/flutter/flutter/issues/6751#issuecomment-730028574.